### PR TITLE
ImmersedBoundaryCondition for drag, take 2

### PR DESCRIPTION
**Pull request by simone-silvestri**
This PR is a cleaner version of #130, removing the forcing for drag above topography in favor of `ImmersedBoundaryCondition` 

closes #130

### CODE REVIEW

File: src/OceanSimulations/OceanSimulations.jl [hunk 1]
```diff
@@ -65,26 +62,32 @@ function ocean_simulation(grid; Δt = 5minutes,
                           tracer_advection = default_tracer_advection(),
                           verbose = false)
 
-    # Set up boundary conditions using Field
+    # allocate memory in `Field`s for surface forcing boundary conditions
```
Comment (glwagner) [inline src/OceanSimulations/OceanSimulations.jl:65]:
```suggestion
    # Allocate memory in `Field`s for surface forcing boundary conditions
```

**Review by glwagner — APPROVED** [review]

File: Project.toml [hunk 2]
```diff
@@ -39,7 +39,7 @@ ImageMorphology = "0.4"
 JLD2 = "0.4, 0.5"
 KernelAbstractions = "0.9"
 NCDatasets = "0.12, 0.13, 0.14"
-Oceananigans = "0.91.13"
+Oceananigans = "0.93.1"
```
Comment (glwagner) [inline Project.toml:42]:
```suggestion
Oceananigans = "0.93.1 - 0.99"
```

**Comment by glwagner** [comment]
@simone-silvestri can you push this through

**Comment by simone-silvestri** [comment]
There seems to be a problem with the example. I will try it locally

**Pull request merged**